src/main/scala/org/ergoplatform/mining/difficulty/DifficultyAdjustment.scala

```diff
@@ -33,8 +32,57 @@ class LinearDifficultyControl(val chainSettings: ChainSettings) extends ScorexLo
     }
   }
 
+  /** @param previousHeaders  should be last headers of the previous epochs */
+  private def bitcoinCalculate(previousHeaders: Seq[Header], epochLength: Int): Difficulty = {
+    val hs = previousHeaders.takeRight(2)
+    bitcoinCalculate(hs(0), hs(1), epochLength: Int)
+  }
+
+  /**
+    * Calculate difficulty as done in Bitcoin (with no capping result)
+    *
+    * Please note this method does not normalize its result!
+    *
+    * @param start - last block of previous epoch
+    * @param end - last block of current epoch
+    */
+  private def bitcoinCalculate(start: Header, end: Header, epochLength: Int): BigInt = {
+    end.requiredDifficulty * desiredInterval.toMillis * epochLength / (end.timestamp - start.timestamp)
+  }
+
+  /**
+    * Calculate difficulty for first block of a new epoch according to EIP-37
+    * @param previousHeaders - last headers of few epochs (8 in case of Ergo mainnet)
+    * @param epochLength - epoch length
+    */
+  def eip37Calculate(previousHeaders: Seq[Header], epochLength: Int): Difficulty = {
+    require(previousHeaders.size >= 2, "at least two headers needed for diff recalc")
+    val lastDiff = previousHeaders.last.requiredDifficulty
+
+    val predictiveDiff = calculate(previousHeaders, epochLength)
+    val limitedPredictiveDiff = if (predictiveDiff > lastDiff) {
+      predictiveDiff.min(lastDiff * 3 / 2)
+    } else {
+      predictiveDiff.max(lastDiff * 2)
```

Comment:
Should it be `lastDiff / 2`? The same goes for `uncompressedDiff`.

Reply:
Sorry, my bad, fixed.

For tests, thinking to add some test vectors. Anything better we can do?

Reply:
PBT might be a good fit here.